internal/execute/tableutil/arrow_builder.go

```diff
@@ -0,0 +1,116 @@
+package tableutil
+
+import (
+	"github.com/apache/arrow/go/arrow/array"
+	"github.com/apache/arrow/go/arrow/memory"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/arrow"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/internal/errors"
+)
+
+// ArrowBuilder is a Builder that uses arrow array builders
+// as the underlying builder mechanism.
+type ArrowBuilder struct {
+	GroupKey  flux.GroupKey
+	Columns   []flux.ColMeta
+	Builders  []array.Builder
```

Comment:
My rationale for this name was because other implementation don't generally specifically mention arrow, but this exposes the arrow builders. So it is a table builder that uses arrow. The original name was `ArrowTableBuilder` but I wanted to avoid the stutter of `tableutil.ArrowTableBuilder`.